wiki/multilang_md.py

```diff
@@ -0,0 +1,333 @@
+                        else:  # 2~
+                            lmin = int(level[0])
+                    else:  # 2
+                        lmin, lmax = int(level)
+                    # toc emoji
+                    no_emoji_re = re.compile("no-emoji")
+                    enable_emoji = no_emoji_re.search(self.doc.last_line) is None
+                    # toc record
+                    for all_doc in self.lang_doc.values():
+                        all_doc.set_toc(
+                            _min=lmin, _max=lmax, enable_emoji_header=enable_emoji
+                        )
+                # detect signal
+                elif self.pattern.match(self.doc.last_line):
+                    keyword_detected = True
+                    signal = self.doc.last_line.replace(" ", "")[5:-5]
+            # write
+            if not keyword_detected:
+                if signal == "common":
+                    for all_doc in self.lang_doc.values():
+                        all_doc.write(self.doc.last_line, codeblock_mark)
+                elif signal == "ignore":
+                    pass
+                else:
+                    self.lang_doc[signal].write(self.doc.last_line, codeblock_mark)
+            # terminal conditions
+            # base doc end
+            if not self.doc.readline():
+                break
+
+        for md in self.lang_doc.values():
+            md.success = True
+
+
+def is_base_md(filename):
+    base = re.compile(r"[.]base[.]md")
+    return base.search(filename)
+
+
+def filtered_base_list(filelist):
+    for filename in filenames:
+        if is_base_md(filename):
+            yield filename
+
+
+def search(dir_name):
+    for path, folder, files in os.walk(dir_name):
+        for file_name in files:
+            if is_base_md(file_name):
+                yield (path, file_name)
+
+
+@click.command()
+@click.argument("filenames", nargs=-1, type=click.Path(exists=True))
+@click.option(
+    "-r",
+    "--recursive",
+    is_flag=True,
+    help="This recursive option searches all subfolders based on current directory and convert all base files.",
+)
+def cli(filenames, recursive):
+    if recursive or filenames:
+        base_count = 0
+        click.secho("----------------------", fg="cyan")
+        if recursive:
+            for path, filename in search("."):
+                MultilangualDoc(path, filename)
+                base_count += 1
+        if filenames:
+            for filename in filtered_base_list(filenames):
+                MultilangualDoc(".", filename)
+                base_count += 1
+        click.secho("----------------------", fg="cyan")
+        click.secho(f" => {base_count} base markdowns were converted.\n", fg="cyan")
+    else:
+        raise click.UsageError(
+            "You have not entered anything. Do 'mmg Foo.base.md' or 'mmg --recursive'."
+        )
+
+
+if __name__ == "__main__":
+    assert sys.version_info[0] == 3
+    try:
+        cli()
```

Comment:
Codacy found an issue: [No value for argument 'filenames' in function call](https://app.codacy.com/gh/mommel/hs-lr-midi-schwurbler/pullRequest?prid=6419091)